misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt

```diff
@@ -866,6 +868,54 @@ class ReflectionQueryFactoryTest {
     }
   }
 
+  @Test
+  fun fetchActorEagerly() {
+    val actorName = "Jeff Goldblum"
+    val characterName = "Ian Malcolm"
+
+    transacter.transaction { session ->
+      val jp = session.save(DbMovie("Jurassic Park", LocalDate.of(1993, 6, 9)))
+      val jg = session.save(DbActor(actorName, LocalDate.of(1952, 10, 22)))
+      session.save(DbCharacter(characterName, session.load(jp), session.load(jg)))
+    }
+
+    val character = transacter.transaction { session ->
+      queryFactory.newQuery<CharacterQuery>()
+        .name(characterName)
+        .withActor()
+        .uniqueResult(session)
+    }
+
+    assertThat(character?.actor?.name).isEqualTo(actorName)
+  }
+
+  @Test
+  fun fetchActorWithoutSpecifyingEager() {
+    val characterName = "Ian Malcolm"
+
+    transacter.transaction { session ->
+      val jp = session.save(DbMovie("Jurassic Park", LocalDate.of(1993, 6, 9)))
+      val jg = session.save(DbActor("Jeff Goldblum", LocalDate.of(1952, 10, 22)))
+      session.save(DbCharacter(characterName, session.load(jp), session.load(jg)))
+    }
+
+    val character = transacter.transaction { session ->
+      queryFactory.newQuery<CharacterQuery>()
+        .name(characterName)
+        .uniqueResult(session)
+    }
+
+    val exception = assertThrows(LazyInitializationException::class.java) {
+      character?.actor?.name
+    }
+
+    assertThat(exception).isNotNull
+    assertThat(exception.message).isNotNull
+
+    val errorMessageRegex = "^could not initialize proxy .* - no Session".toRegex()
```

Comment:
Take it or leave it: omit this? Might make the test more flaky on Hibernate upgrades without improving coverage much